### DISCUSSION TITLE
`props.class` vs `$attrs.class`

### DISCUSSION THREAD

**Discussion by kylemilloy** [unovue/shadcn-vue#606] [General]
Hey I notice in a lot of components we use `props.class` and pass that to a `cn` helper. I get the sense this is a holdover from converting this from react to vue where we have to explicitly define a class prop for components however in vue we can leverage `$attr.class`.

Is my assumption correct or is there another reason we're explicitly defining a class prop? If that is the reason, can we reduce some boilerplate by using `$attr.class` instead?

**Comment by sadeghbarati** [comment]
Reason we used class as props is mentioned in the PR link 

https://github.com/radix-vue/shadcn-vue/pull/241
